src/transformState.js

```diff
@@ -1,11 +1,49 @@
 'use strict';
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- */
+// function transformState(state, actions) {
+//   for (const action of actions) {
+//     if (action.type === 'clear') {
+//       for (const key in state) {
+//         delete state[key];
+//       }
+//     }
+
+//     if (action.type === 'addProperties') {
+//       Object.assign(state, action.extraData);
+//     }
+
+//     if (action.type === 'removeProperties') {
+//       for (const key of action.keysToRemove) {
+//         delete state[key];
+//       }
+//     }
+//   }
+
+//   return state;
+// }
+
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can descturcture the action to get all the needed props of it (extraData, keysToRemove, type)